### ISSUE TITLE
Add option to use GraalJVM

### ISSUE THREAD

**Issue by 4nyNoob** [PrismLauncher/PrismLauncher#2949]
### Role

I play and work making minecraft assets/mods

### Suggestion

add the option to download graaljvm for the java distribution on the auto-download window

### Benefit

graalJVM runs a whole lot better on my PC (R9 5900x)

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

with my limited testing i saw a doubling in performance when generating the world

i don't know if it's a general increase or if it's dependent on computer specs or situational, i would need to run some more formal tests, if anyone has been using graal too i would love to know, since i also switched to it for my mod development

**Comment by maskersss** [comment]
in speedrunning, graalvm is known to be slower than openjdk right after launching mc, but after some time for both of them to warm up graalvm becomes faster than openjdk
(performance here is measured in how many new worlds can be created* per unit of time)
this is also what i saw when testing

**Comment by he3als** [comment]
Added support [to the launcher in my fork](https://github.com/he3als/PrismLauncher/commit/94a191c4be732ae9bb8d12bdfcab1442564e5521), which could be used if this is implemented

If anyone wants to complete this though, the other half of the work (to my knowledge) is adding support to [meta](https://github.com/PrismLauncher/meta). You can use the [Oracle docs](https://www.oracle.com/java/technologies/jdk-script-friendly-urls/) for reference

**Comment by Vanqy9316** [comment]
i would like to extend on what OP said about performance... with the correct flags on a non-memory constrained system, graalVM is always faster. further, the support for the ZGC as an alternative to the old G1C1 has been added recently for java 21, which is a big deal.

**Comment by Mick235711** [comment]
> If anyone wants to complete this, though, the other half of the work (to my knowledge) is adding support to [meta](https://github.com/PrismLauncher/meta). You can use the [Oracle docs](https://www.oracle.com/java/technologies/jdk-script-friendly-urls/) for reference

I also wish Prism Launcher could support GraalVM (and Semeru OpenJ9, OpenJDK, and even Oracle Java, maybe, but at that point, we need to ask how much is too much). However, supporting this is a decidedly non-trivial task.

Backgrounds:
1. GraalVM does not expose a version API like Adoptium (Temurin) and Azul Zulu do, so we cannot get all the released versions with a simple API GET request.
2. The possible solution to this problem is closely connected to the other significant problem: Which version should we support? GraalVM comes in two versions: Community Edition (CE) and Enterprise Edition (EE). Both are free to use for personal use, but only the former is open-sourced. If we only want to support CE, then that has a [GitHub repo](https://github.com/graalvm/graalvm-ce-builds/releases/), and GitHub has an API to get all the releases in a repo so that it will be pretty straightforward. However, EE generally contains much more optimization and performs better in most tasks (incl. Minecraft), so most gamers prefer it.
3. Before June 2023, GraalVM EE is a commercial product with a commercial release, and its downloads are protected under an Oracle login page, so supporting them is a non-starter. However, fortunately, in June 2023, GraalVM EE was [confusingly renamed](https://www.jvm-weekly.com/p/graalvm-ee-is-dead-long-live-oracle) to "Oracle GraalVM" and transitioned to an [open license](https://www.oracle.com/downloads/licenses/graal-free-license.html) that allows free use even in commercial environments. After this transition, all the GraalVM EE versions (named by year) except the two LTS versions (20/21) are discontinued, and new Oracle GraalVM versions used version numbers based on JDK version (so Oracle GraalVM 17, 20, 21, 22, 23, see the confusing [Release Calendar page](https://www.graalvm.org/release-calendar/)). These new versions have open links to download, but they don't have an API to tell you what the released versions *are*.
4. Instead, one possible way (that was adopted by Homebrew) is to scrap the [downloads page](https://www.oracle.com/java/technologies/downloads/#graalvmjava23) to find what versions are available (additionally, we may want to scrap the [archive page](https://www.oracle.com/java/technologies/javase/graalvm-jdk21-archive-downloads.html) to get previous versions). This is more work (but probably we can use the same regex as Homebrew used), but I think it may be worth supporting. If we can support GraalVM EE (i.e., Oracle GraalVM), I don't think we need to support GraalVM CE.

**Edit**: I just discovered that Prism Launcher actually expects the *release date* of runtimes to be present; this is not available on the download/archive page, so the only solution is to scrap the [Release Calendar page](https://www.graalvm.org/release-calendar/), and hope that Oracle updates this page as soon as a new version is released... this makes me hesitant about this support.

So basically, at the end, there are three questions that need answering:
- GraalVM does not provide JRE, only JDK. Is that acceptable, given all currently supported releases/vendors are JREs?
- Is "~~download page~~ release calendar scraping" an acceptable (unofficial) way of getting all the released versions?
- Which Java vendor do we want to support besides Mojang (i.e. Microsoft), Temurin, and Zulu?
    - Possible answer: GraalVM, IBM Semeru (the OpenJ9 VM release), OpenJDK, Oracle JDK, Amazon Corretto, BellSoft Liberica

If there is significant interest, I think I can try to put up a PR to meta to try to add the necessary release JSON files.

**Comment by 4nyNoob** [comment]
it would be a huuuge help to get GraalVM(EE) in such an easy manner :D

I personally believe that the regex method is completely acceptable if we get a clear error message for tbe user who fails to download Graal because the regex is broken, **if** it breaks in the future
